src/wallet/rpcwallet.cpp

```diff
@@ -1412,7 +1412,11 @@ static RPCHelpMan listtransactions()
                     {"label|dummy", RPCArg::Type::STR, RPCArg::Optional::OMITTED_NAMED_ARG, "If set, should be a valid label name to return only incoming transactions\n"
                           "with the specified label, or \"*\" to disable filtering and return all transactions."},
                     {"count", RPCArg::Type::NUM, RPCArg::Default{10}, "The number of transactions to return"},
-                    {"skip", RPCArg::Type::NUM, RPCArg::Default{0}, "The number of transactions to skip"},
+                    {"options|skip", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED_NAMED_ARG, "for backward compatibility: passing in a number instead of an object will result in {\"skip\":nnn}",
```

Comment:
I'm not sure here, but previous reviews on questions like this and `doc/developer-notes.md` generally seemed to discourage mixing types / overloading:

```
- Try not to overload methods on argument type. E.g. don't make `getblock(true)`
  and `getblock("hash")` do different things.
```

Reply:
That's about supported use, not deprecation compatibility...

Reply:
Shouldn't the old use be deprecated with planned removal then?